src/oss/langchain-messages.mdx

```diff
@@ -2,14 +2,488 @@
 title: Messages
 ---
 
+Messages are the fundamental unit of communication in LangChain chat models. They represent the input and output of conversations, carrying both content and metadata that enables rich, multimodal interactions.
+
+TODO link out to chat model conceptual guide
+
 ## Overview
 
+Messages are structured objects that contain:
+
+- **Role**: Identifies the message type (`system`, `user`, `assistant`, or `tool`)
+- **Content**: Types of content: text, images, audio, documents, and more
+- **Metadata**: Optional fields like IDs, names, usage information, and response metadata from providers
+
+LangChain provides a unified message format that works across all chat model providers, handling provider-specific differences automatically.
+
+### Message Types
+
+| Type | Role | Purpose |
+|------|------|---------|
+| [SystemMessage](#system-message) | `system` | Prime model behavior and provide context |
+| [HumanMessage](#human-message) | `user` | Represent user input and interactions |
+| [AIMessage](#ai-message) | `assistant` | Model responses and tool requests |
+| [ToolMessage](#tool-message) | `tool` | Tool execution results |
+
+### Basic Usage
+
+TODO model switcher
+
+```python
+from langchain_core.messages import HumanMessage, AIMessage, SystemMessage
+from langchain_openai import ChatOpenAI
+
+model = ChatOpenAI(model="gpt-oss", output_version="v1")
+
+# Create messages
+system_msg = SystemMessage("You are a helpful assistant.")
+human_msg = HumanMessage("Hello, how are you?")
+ai_msg = AIMessage("I'm doing well, thank you!")
+
+# Use with chat models
+messages = [system_msg, human_msg]
+response = model.invoke(messages)  # Returns AIMessage
+
+# Or, more simply:
+response = model.invoke("Hello, how are you?")
+```
+
+### OpenAI Format Compatibility
+
+TODO does this deserve its own section here? or just a brief note?
+
+LangChain messages are compatible with OpenAI's format:
```

Comment:
this is a bit weird bc it shows compatibility with models? not between message format

do we have utils for converting from one to another that we want to publically expose?